### DISCUSSION TITLE
Implementing Insert/RemoveColumn/Row methods as voids besides function?

### DISCUSSION THREAD

**Discussion by cmoha** [mathnet/mathnet-numerics#826] [Q&A] [answered]
Are there any reasons for methods such as `InsertColumn, RemoveColumn, InsertRow, RemoveRow` not be implemented as void also?

**Comment by cdrnet** [comment, marked as answer]
You mean as inplace operations? Mainly because matrices cannot change their dimension after creation. This is by design, but ultimately because even for sparse matrices all of these operations require to allocate and copy to new internal data structures (mostly arrays).

If you need to insert and remove rows or columns often, do you really need a full matrix or would a list of vectors maybe be more appropriate?
